scrape/scrape.go

```diff
@@ -1797,6 +1825,60 @@ func (sl *scrapeLoop) checkAddError(met []byte, err error, sampleLimitErr, bucke
 	}
 }
 
+// tryAppendOMCTZeroSample tries to append a created timestamp sample to the appender.
+// The OpenMetrics parser is used to parse the _created line, and, due to the behavior
+// of OpenMetrics-text, we need to parse the next line to verify if it's a valid
+// created timestamp or not.
+//
+// This particular behavior can potentially have side effects on the scrape loop.
+// 1) If we try to parse the next line and we find EOF we need to stop the loop, but not
+// before finishing the append of the regular sample. This behavior can be checked by verifying
+// the returned error.
+// 2) If the next line is not a created timestamp, in the next loop we need to skip parsing
+// the next line since we're already in the right position for the next iteration.
+//
+// The function returns a boolean indicating if we should skip calling the Next() method
+// in the next iteration.
+func (sl *scrapeLoop) tryAppendOMCTZeroSample(p *textparse.OpenMetricsParser,
```

Comment:
Yea, this could work but is too convoluted. I believe there is a cleaner way of doing this. Let's brainstorm in https://github.com/prometheus/prometheus/pull/13506/files#r1506303440